pkg/ui/workspaces/cluster-ui/src/databasesPage/databasesPage.tsx

```diff
@@ -536,111 +539,124 @@ export class DatabasesPage extends React.Component<
+          <Tooltip
+            placement="bottom"
+            title="The total number of tables in the database."
+          >
+            Tables
+          </Tooltip>
+        ),
+        cell: database =>
+          this.checkInfoAvailable(database, database.tableCount),
+        sort: database => database.tableCount,
+        className: cx("databases-table__col-table-count"),
+        name: "tableCount",
+      },
+      {
+        title: (
+          <Tooltip
+            placement="bottom"
+            title="The total number of ranges across all tables in the database."
+          >
+            Range Count
+          </Tooltip>
+        ),
+        cell: database =>
+          this.checkInfoAvailable(database, database.rangeCount),
+        sort: database => database.rangeCount,
+        className: cx("databases-table__col-range-count"),
+        name: "rangeCount",
+      },
+      {
+        title: (
+          <Tooltip
+            placement="bottom"
+            title="Regions/Nodes on which the database tables are located."
+          >
+            {this.props.isTenant ? "Regions" : "Regions/Nodes"}
+          </Tooltip>
         ),
-      sort: database => database.nodesByRegionString,
-      className: cx("databases-table__col-node-regions"),
-      name: "nodeRegions",
-      hideIfTenant: true,
-    },
-    {
-      title: (
-        <Tooltip
-          placement="bottom"
-          title="Index recommendations will appear if the system detects improper index usage, such as the
+        cell: database =>
+          this.checkInfoAvailable(
+            database,
+            database.nodesByRegionString || "None",
+          ),
+        sort: database => database.nodesByRegionString,
+        className: cx("databases-table__col-node-regions"),
+        name: "nodeRegions",
+        hideIfTenant: true,
+      },
+    ];
+    if (this.props.indexRecommendationsEnabled) {
+      columns.push({
+        title: (
+          <Tooltip
+            placement="bottom"
+            title="Index recommendations will appear if the system detects improper index usage, such as the
           occurrence of unused indexes. Following index recommendations may help improve query performance."
-        >
-          Index Recommendations
-        </Tooltip>
-      ),
-      cell: this.renderIndexRecommendations,
-      sort: database => database.numIndexRecommendations,
-      className: cx("databases-table__col-idx-rec"),
-      name: "numIndexRecommendations",
-    },
-  ];
+          >
+            Index Recommendations
+          </Tooltip>
+        ),
+        cell: this.renderIndexRecommendations,
+        sort: database => database.numIndexRecommendations,
+        className: cx("databases-table__col-idx-rec"),
+        name: "numIndexRecommendations",
+      });
+    }
+    return columns;
+  }
 
   render(): React.ReactElement {
-    this.columns.find(c => c.name === "nodeRegions").showByDefault =
+    const columns = this.columns();
```

Comment:
Ideally we can compute this only when dependencies change, which looks like it might just be the `indexRecommendationsEnabled` prop that changes the columns.

Reply:
Add the columns are part of state, recompute columns when `indexRecommendationsEnabled` changes in `componentDidUpdate`